pyrealm/core/calendar.py

```diff
@@ -118,3 +118,78 @@ def __repr__(self) -> str:
         """Representation of a Calendar instance."""
 
         return f"Calendar({self.dates[0]!s}, {self.dates[-1]!s})"
+
+
+@dataclass
+class LocationDateTime:
+    """A data class representing an observation location and date and time information.
+
+    This class encapsulates the latitude and longitude of a location along with a
+    corresponding time array. It automatically calculates the latitude and longitude in
+    radians, the Julian days from the date-time information, and a decimal
+    representation of the local time.
+
+    Example:
+        >>> import numpy as np
+        >>> ldt = LocationDateTime(
+        ...     latitude=-35.058333,
+        ...     longitude=147.34167,
+        ...     UTC_offset=10,
+        ...     year_date_time=np.array([np.datetime64("2024-08-12T10:30")])
```

Comment:
It isn't clear from this example and docs that the `year_date_time` is assumed to be local time. Is UTC optional?

Reply:
This is an oversight on my part. I've removed the references to UTC offset. They were a hangover from when I thought calculation of solar elevation required knowledge of local time zone.